python/ambassador/diagnostics/diagnostics.py

```diff
@@ -118,8 +118,7 @@ def __init__(self, diag: "Diagnostics", estat: EnvoyStats, request) -> None:
 
         # Go ahead and grab Envoy cluster stats for all possible clusters.
         # XXX This might be a bit silly.
-        self.cluster_names = [cluster.envoy_name for cluster in self.diag.clusters.values()]
-        self.cstats = {name: self.estat.cluster_stats(name) for name in self.cluster_names}
+        self.cstats = {cluster.envoy_name: self.estat.cluster_stats(cluster.stats_name) for cluster in self.diag.clusters.values()}
```

Comment:
Can you add to the PR description a sample of the before and after for `cstats` here? I think I'm getting tripped up due to the fact that the cstats name is being reused in DiagResults and EnvoyStats so having the diff will help it make sense.

Just for information:

`envoy_name` is the cluster name but potentially mangled. The reason for this is that Envoy only allows 60 characters for the name of a cluster. So, we have logic that where there are really long names we will truncate them and handle name collisions. This mangled name is what is stored here and ultimately passed on to Envoy configuration.

Not sure if that matters but just wanted to call that out.